src/hotspot/os_cpu/linux_aarch64/vm_version_linux_aarch64.cpp

```diff
@@ -167,3 +167,22 @@ void VM_Version::get_os_cpu_info() {
     fclose(f);
   }
 }
+
+void VM_Version::get_compatible_board(char *buf, int buflen) {
+  assert(buf != NULL, "invalid argument");
+  *buf = '\0';
+  int fd = open("/proc/device-tree/compatible", O_RDONLY);
+  if (fd != -1) {
+    struct stat statbuf;
+    fstat(fd, &statbuf);
+    ssize_t read_sz = read(fd, buf, statbuf.st_size);
```

Comment:
This looks wrong: the read() call should use buflen.

Reply:
Fixed it in new commit.